### DISCUSSION TITLE
Modifying a Network

### DISCUSSION THREAD

**Discussion by Bankst** [xR3b0rn/dbcppp#146] [Q&A]
I have successfully loaded a few different DBC files and walked through them, but for my application I also want to be able to modify and save these. The README.md claims writing is possible but I see no way to add/remove Messages/Nodes to a Network, nor add/remove Signals to Messages.

Am I missing something or is this simply not supported?

**Comment by Bankst** [comment]
@xR3b0rn I've been trying to figure this out for a while and the best I can tell is I'm intended to re-create the INetwork/ISignals every time I want to modify them for saving. Is this the intended use?

**Reply by xR3b0rn** [reply]
The library was never created with the intention of manipulating a DBC. If you still want to do this, you may need to customize the library with setter/adder functions.

**Reply by Bankst** [reply]
It looks like I could just make the NetworkImpl/SignalImpl std::vectors public and modify those, if I am reading this right.

**Reply by xR3b0rn** [reply]
Could work.
